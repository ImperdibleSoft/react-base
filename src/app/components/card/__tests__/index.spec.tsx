import * as React from 'react';
import { shallow } from 'enzyme';

import { Card } from '..';

describe('Card', () => {
  it('should render correctly a populated card', () => {
    const component = shallow(
      <Card>
        <p>test</p>
      </Card>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly an empty card', () => {
    const component = shallow(<Card />);

    expect(component).toMatchSnapshot();
  });
});
