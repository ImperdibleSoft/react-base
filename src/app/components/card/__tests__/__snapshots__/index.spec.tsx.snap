// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly a populated card 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "card__Card-sc-1hrkoho-0",
        "isStatic": false,
        "rules": Array [
          "background-color:",
          [Function],
          ";border-radius:4px;display:inline-block;margin:5px;padding:10px;text-align:left;transition:background-color 0.2s;vertical-align:top;width:calc(100% - 10px);",
          "@media screen and (min-width: 720px)",
          "{margin:10px;padding:20px;width:",
          [Function],
          ";}",
        ],
      },
      "displayName": "Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "card__Card-sc-1hrkoho-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <p>
    test
  </p>
</StyledComponent>
`;

exports[`Card should render correctly an empty card 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "card__Card-sc-1hrkoho-0",
        "isStatic": false,
        "rules": Array [
          "background-color:",
          [Function],
          ";border-radius:4px;display:inline-block;margin:5px;padding:10px;text-align:left;transition:background-color 0.2s;vertical-align:top;width:calc(100% - 10px);",
          "@media screen and (min-width: 720px)",
          "{margin:10px;padding:20px;width:",
          [Function],
          ";}",
        ],
      },
      "displayName": "Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "card__Card-sc-1hrkoho-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
